File: frontend/src/lib/currency.ts
// Валюты без привязки к курсу: каждая цена хранит свой код.
// Дефолтный код приходит из настроек (/settings), символы — отсюда.

export const CURRENCY_SYMBOLS: Record<string, string> = {
  USD: "$",
  EUR: "€",
  RUB: "₽",
  GBP: "£",
  CNY: "¥",
};

export const SUPPORTED_CURRENCIES = Object.keys(CURRENCY_SYMBOLS);

export const DEFAULT_CURRENCY = "USD";

export function currencySymbol(code: string | null | undefined): string {
  if (!code) return CURRENCY_SYMBOLS[DEFAULT_CURRENCY];
  return CURRENCY_SYMBOLS[code.toUpperCase()] || code;
}

// "12 000 $" — целое с неразрывными пробелами + символ валюты
export function formatPrice(
  value: number | string | null | undefined,
  code: string | null | undefined
): string {
  if (value == null || value === "") return "—";
  const num = typeof value === "string" ? Number(value) : value;
  if (Number.isNaN(num)) return "—";
  return `${Math.round(num).toLocaleString("ru")} ${currencySymbol(code)}`;
}

File: backend/app/currency.py
"""Поддерживаемые валюты и символы.

Без привязки к курсу: каждая цена/сделка хранит свой код валюты, конвертации нет.
Глобальный дефолт (admin) лежит в app_settings под ключом `default_currency`
и используется как стартовое значение в формах.
"""

DEFAULT_CURRENCY = "USD"

# Код → символ. Один источник правды для бэка/PDF (фронт и бот дублируют тот же список).
CURRENCY_SYMBOLS: dict[str, str] = {
    "USD": "$",
    "EUR": "€",
    "RUB": "₽",
    "GBP": "£",
    "CNY": "¥",
}

SUPPORTED_CURRENCIES: list[str] = list(CURRENCY_SYMBOLS.keys())


def normalize_currency(value: str | None) -> str:
    """Привести к поддерживаемому коду, иначе вернуть дефолт."""
    if not value:
        return DEFAULT_CURRENCY
    code = value.strip().upper()
    return code if code in CURRENCY_SYMBOLS else DEFAULT_CURRENCY


def symbol(code: str | None) -> str:
    return CURRENCY_SYMBOLS.get(normalize_currency(code), "$")


def format_price(value, code: str | None) -> str:
    """`12000 $` — целое с пробелами-разделителями и символом валюты."""
    if value is None:
        return "—"
    try:
        amount = f"{int(float(value)):,}".replace(",", " ")
    except (TypeError, ValueError):
        return "—"
    return f"{amount} {symbol(code)}"
